src/app/utils/field-validation/field-validation.utils.ts

```diff
@@ -0,0 +1,95 @@
+import { isEqual } from 'lodash'
+import { CLIENT_CHECKBOX_OTHERS_INPUT_VALUE } from 'shared/constants'
+
+import { BasicField } from '../../../../shared/types'
+import { ParsedClearFormFieldResponseV3 } from '../../../types/api/submission'
+
+import { isGenericStringAnswerResponseV3 } from './field-validation.guards'
+
+export const checkIsResponseChangedV3 = ({
+  response,
+  prevResponse,
+}: {
+  response: ParsedClearFormFieldResponseV3
+  prevResponse?: ParsedClearFormFieldResponseV3
+}): boolean => {
+  if (!prevResponse) {
+    return true
+  }
+  if (response.fieldType !== prevResponse.fieldType) {
+    return true
+  }
+
+  if (isGenericStringAnswerResponseV3(response)) {
+    return response.answer.toString() !== prevResponse.answer.toString()
+  }
+
+  switch (response.fieldType) {
+    case BasicField.YesNo:
+      return response.answer !== prevResponse.answer
+    case BasicField.Email:
+    case BasicField.Mobile:
+      return !(
+        response.fieldType === prevResponse.fieldType &&
+        prevResponse.answer.value === response.answer.value &&
+        prevResponse.answer.signature === response.answer.signature
+      )
+    case BasicField.Radio:
+      if (prevResponse.fieldType !== response.fieldType) {
+        return true
+      }
+      // eslint-disable-next-line no-case-declarations
+      const prevResponseValue =
+        'value' in prevResponse.answer
+          ? prevResponse.answer.value
+          : 'othersInput' in prevResponse.answer
+            ? prevResponse.answer.othersInput
+            : null
+      // eslint-disable-next-line no-case-declarations
+      const responseValue =
+        'value' in response.answer
+          ? response.answer.value
+          : 'othersInput' in response.answer
+            ? response.answer.othersInput
+            : null
+      return prevResponseValue !== responseValue
+    case BasicField.Checkbox:
+      if (prevResponse.fieldType !== response.fieldType) {
```

Comment:
this is required due to TS throwing an error as it fails to infer the type of prevResponse is same as response without it, despite being checked in L19 above.